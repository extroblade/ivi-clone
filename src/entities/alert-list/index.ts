export { AlertList } from './ui/alert-list';
