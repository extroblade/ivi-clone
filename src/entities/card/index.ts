export type { CardProps } from './model/props';
export { Card } from './ui/card';
export { CardLoader } from './ui/card-loader';
export { ShowAllCard } from './ui/show-all-card';
