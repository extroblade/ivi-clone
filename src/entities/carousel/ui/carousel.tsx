import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';

import Link from 'next/link';
import { FC } from 'react';
import { MdArrowForwardIos } from 'react-icons/md';
import Slider from 'react-slick';

import { Title } from '@/newui';

import { defaultSettings } from '../model/default-settings';
import { CarouselProps } from '../model/props';
import styles from './carousel.module.scss';

export const Carousel: FC<CarouselProps> = ({ title, children, route, settings }): JSX.Element => {
  return (
    <div className={styles.carousel}>
      {title && (
        <div className={styles.title}>
          <Link title={title} href={route || ''} className={styles.title_container}>
            <Title tag={'h4'}>{title}</Title>
            <MdArrowForwardIos />
          </Link>
        </div>
      )}
      <Slider {...(settings || defaultSettings)}>{children}</Slider>
    </div>
  );
};
