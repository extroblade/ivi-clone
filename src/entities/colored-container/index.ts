export { ColorContainer } from './ui/color-container';
