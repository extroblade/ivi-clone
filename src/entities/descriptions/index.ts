export { AnimationDescription } from './animation';
export { MainDescription } from './main';
export { MoviesDescription } from './movies';
export { SeriesDescription } from './series';
