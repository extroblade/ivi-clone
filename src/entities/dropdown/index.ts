export { ChooseDropdown } from './choose/ui/ChooseDropdown';
export { SearchDropdown } from './search/ui/SearchDropdown';
export { SortDropdown } from './sort/ui/SortDropdown';
export { Dropdown } from './ui/Dropdown';
