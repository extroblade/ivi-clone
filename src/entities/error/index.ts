export { Error as FetchError } from './ui/error';
