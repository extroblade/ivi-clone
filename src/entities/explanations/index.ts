export { Explanations } from './ui/Explanations';
