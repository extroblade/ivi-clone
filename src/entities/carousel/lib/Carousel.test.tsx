import { render } from '@testing-library/react';
import { Provider } from 'react-redux';

import { Carousel } from '@/entities/carousel';
import { mockStore } from '@/shared/testdata';

describe('carousel', () => {
  test('number array', () => {
    const carousel = render(
      <Provider store={mockStore}>
        <Carousel title={'title'}> </Carousel>
      </Provider>
    );
    expect(carousel).toMatchSnapshot();
  });
});
