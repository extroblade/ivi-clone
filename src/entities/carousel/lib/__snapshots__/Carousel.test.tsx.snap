// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`carousel number array 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="carousel"
      >
        <div
          class="title"
        >
          <a
            class="title_container"
            href=""
            title="title"
          >
            <h4
              class="h h4"
            >
              title
            </h4>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
              />
            </svg>
          </a>
        </div>
        <div
          class="slick-slider slick-initialized"
        >
          <div
            class="slick-list"
          >
            <div
              class="slick-track"
              style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="carousel"
    >
      <div
        class="title"
      >
        <a
          class="title_container"
          href=""
          title="title"
        >
          <h4
            class="h h4"
          >
            title
          </h4>
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6.23 20.23L8 22l10-10L8 2 6.23 3.77 14.46 12z"
            />
          </svg>
        </a>
      </div>
      <div
        class="slick-slider slick-initialized"
      >
        <div
          class="slick-list"
        >
          <div
            class="slick-track"
            style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
